tests/subscribe/useSubscription.spec.ts

```diff
@@ -366,4 +394,43 @@ describe('subscribe', () => {
 
   })
 
+  it('it does not return undefined when a subscription changes', async () => {
+    jest.useFakeTimers()
+
+    function action(value: number): Promise<number> {
+      return new Promise((resolve) => setTimeout(() => resolve(value), 100))
+    }
+
+    const originalValue = 0
+    const valueArg = ref(originalValue)
+    const subscription = useSubscription(action, [valueArg])
+
+    jest.runAllTimers()
+    jest.useRealTimers()
+
+    await timeout()
+
+    valueArg.value = 1
+
+    await timeout()
+
+    expect(subscription.response.value).toBe(originalValue)
```

Comment:
This test breaks my brain but I assume it works because `timeout` is flushing at 0ms and even though the value of `valueArg` is changing the subscription channel it proves the original value is still available after the scheduler flush but before the resolution at 100ms?

Reply:
I simplified this in the most recent commit. The timeout on 411 wasn't necessary. But essentially use fake timers and run the timers to execute the action. Then revert back to real timers. With real timers the setTimeout in the asyn action doesn't execute until after the `expect` which means we can verify the response is correct.